src/Components/AmazonPay/ButtonConfiguration.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PayonePayment\Components\AmazonPay;
+
+use PayonePayment\Components\ConfigReader\ConfigReader;
+use RuntimeException;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Struct\ArrayStruct;
+use Shopware\Core\System\Language\LanguageEntity;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+
+class ButtonConfiguration
+{
+    private const DEFAULT_PUBLIC_KEY = 'AE5E5B7B2SAERURYEH6DKDAZ';
+
+    public function __construct(
+        private readonly ConfigReader $configReader,
+        private readonly EntityRepository $languageRepository
+    ) {
+    }
+
+    public function getButtonConfiguration(
+        SalesChannelContext $context,
+        string $location,
+        array $addPayData,
+        bool $isExpress,
+        ?float $totalAmount = null,
+        ?string $currencyIso = null,
+    ): ArrayStruct {
+        $config = $this->configReader->read($context->getSalesChannelId());
+
+        $payloadJson = $addPayData['payload'];
+
+        $payload = json_decode($payloadJson, true);
+        if ($totalAmount === null) {
+            $totalAmount = (float)$payload['paymentDetails']['totalOrderAmount']['amount'];
```

Comment:
Can you be sure that it is always set exactly like this? Should you set a default value or throw an exception if it is not set? (same below)